# Evaluation And Error Analysis

Tune thresholds, inspect false positives and false negatives, and save an error-analysis artifact. This tests richer evaluation loops after a model exists.

In [1]:
from pathlib import Path

exec(Path("samples/ml_tasks/evaluation_error_analysis.py").read_text(encoding="utf-8"))


analysis_artifact: artifacts/ml_tasks/evaluation_error_analysis.json
best_threshold:
- threshold: 0.4000
- tp: 133
- fp: 43
- tn: 80
- fn: 10
- precision: 0.7557
- recall: 0.9301
- accuracy: 0.8008
- f1: 0.8339
- brier: 0.1573
false_positives_reviewed: 8
false_negatives_reviewed: 8


threshold,precision,recall,f1,fp,fn
0.25,0.624,0.986,0.764,85,2
0.30,0.667,0.979,0.793,70,3
0.35,0.704,0.965,0.814,58,5
0.40,0.756,0.930,0.834,43,10
0.45,0.787,0.881,0.832,34,17
0.50,0.809,0.797,0.803,27,29
0.55,0.821,0.706,0.759,22,42
0.60,0.835,0.636,0.722,18,52
0.65,0.874,0.531,0.661,11,67
0.70,0.879,0.406,0.555,8,85


## Finding

The threshold sweep selects **0.40** as the best operating point by F1. At this threshold the model reaches precision **0.7557**, recall **0.9301**, accuracy **0.8008**, and F1 **0.8339**, with **43** false positives and **10** false negatives.

Error analysis reviewed **8 false positives** and **8 false negatives**. The saved run artifact is `artifacts/ml_tasks/evaluation_error_analysis.json`, which should be inspected alongside the threshold table when deciding whether the high-recall 0.40 threshold is acceptable for the workflow.
